Formula/yarn.rb

```diff
@@ -18,9 +16,13 @@ class Yarn < Formula
   depends_on "node"
 
   def install
-    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
-    bin.install_symlink Dir["#{libexec}/bin/*"]
-    inreplace "#{libexec}/lib/node_modules/yarn/package.json", '"installationMethod": "tar"', '"installationMethod": "homebrew"'
+    libexec.install Dir["*"]
+    package_json = JSON.parse File.read("#{libexec}/package.json")
```

Comment:
If we're going down this road of parsing these json packaging files ourselves, which I don't think is advisable, it would belong in the brew backend with a DSL, not proliferated throughout formulae.